### ISSUE TITLE
Kubernetes Controller fails when using a a preexisitng vnet  is used from another ResourceGroup

### ISSUE THREAD

**Issue by ivanthelad** [Azure/acs-engine#538]
Hi, 
While this issue has come up become before. It is not clear if its been resolved.  Has this been fixed in the cloud controller or not?. When this occurs the scheduler fails to schedule pods to nodes 

https://github.com/kubernetes/kubernetes/issues/38134
https://github.com/Azure/acs-engine/issues/120#issuecomment-262757203

0426 11:09:14.421906       1 routecontroller.go:154] Creating route for node k8s-master-18296070-0 10.244.0.0/24 with hint 91673268-2a65-11e7-a50f-000d3a26f4a1, throttled 515ns
I0426 11:09:14.421944       1 azure_routes.go:65] create: creating route. clusterName="kubernetes24" instance="k8s-master-18296070-0" cidr="10.244.0.0/24"
E0426 11:09:14.421582       1 routecontroller.go:160] Could not create route be6617f6-2a65-11e7-a50f-000d3a26f4a1 10.244.2.0/24 for node k8s-agentpool1-18296070-1 after 61.739077ms: network.SubnetsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Network/virtualNetworks/kubernetesvnet' under resource group 'kubernetes24' was not found."
I0426 11:09:14.422456       1 routecontroller.go:154] Creating route for node k8s-agentpool1-18296070-1 10.244.2.0/24 with hint be6617f6-2a65-11e7-a50f-000d3a26f4a1, throttled 619ns
I0426 11:09:14.422476       1 azure_routes.go:65] create: creating route. clusterName="kubernetes24" instance="k8s-agentpool1-18296070-1" cidr="10.244.2.0/24"
E0426 11:09:14.469835       1 routecontroller.go:160] Could not create route be6617f6-2a65-11e7-a50f-000d3a26f4a1 10.244.2.0/24 for node k8s-agentpool1-18296070-1 after 47.377969ms: network.SubnetsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Network/virtualNetworks/kubernetesvnet' under resource group 'kubernetes24' was not found."
E0426 11:09:14.514898       1 routecontroller.go:160] Could not create route 91673268-2a65-11e7-a50f-000d3a26f4a1 10.244.0.0/24 for node k8s-master-18296070-0 after 92.981052ms: network.SubnetsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Network/virtualNetworks/kubernetesvnet' under resource group 'kubernetes24' was not found."
I0426 11:09:14.514926       1 routecontroller.go:154] Creating route for node k8s-master-18296070-0 10.244.0.0/24 with hint 91673268-2a65-11e7-a50f-000d3a26f4a1, throttled 928ns
I0426 11:09:14.514926       1 routecontroller.go:154] Creating route for node k8s-master-18296070-0 10.244.0.0/24 with hint 91673268-2a65-11e7-a50f-000d3a26f4a1, throttled 928ns
I0426 11:09:14.514934       1 azure_routes.go:65] create: creating route. clusterName="kubernetes24" instance="k8s-master-18296070-0" cidr="10.244.0.0/24"
E0426 11:09:14.723881       1 routecontroller.go:160] Could not create route 91673268-2a65-11e7-a50f-000d3a26f4a1 10.244.0.0/24 for node k8s-master-18296070-0 after 208.951798ms: network.SubnetsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Network/virtualNetworks/kubernetesvnet' under resource group 'kubernetes24' was not found."
ivan@k8s-master-18296070-0:~$ kubectl  logs kube-controller-manager-k8s-master-18296070-0 -n kube-system

**Comment by JackQuincy** [comment]
This is a upstream issue and the issue should be filed on kubernetes/kubernetes as well. This is not currently supported. We do have support to deploy your cluster into a vnet in that is peered with a vnet in another resourcegroup. There is some setup required to get that working and if you do take that route you should search the issues here to see how others have done it successfully.

**Comment by colemickens** [comment]
This will be fixed when 1.6 is available. The route controller will no longer try to look  for the route table on the subnet. And then step 7 in the walkthrough will be required to link the routeTable to the existing subnet.

**Comment by seanknox** [comment]
Hi @ivanthelad. Kubernetes 1.6 has landed in master. Can you try building a cluster using acs-engine + 1.6 and see if the RG issue is indeed fixed?

**Comment by jdumars** [comment]
@ivanthelad any feedback on impact of 1.6 on this issue?

**Comment by itaysk** [comment]
Just created a cluster in a legacy VNET in a separate RG. We should close this issue to reflect that for the sake of future visitors

**Comment by justinbarias** [comment]
I've rebuilt using Kubernetes 1.6.2, but still having the same problem:
* K8sresourcegroup is the RG where the cluster is built
* The name of the VNET resource group is kubeshared

I0619 06:52:22.924573       1 routecontroller.go:154] Creating route for node k8s-master-21425678-0 10.244.1.0/24 with hint d1afc54f-54b9-11e7-a827-000d3aa29c74, throttled 700ns
I0619 06:52:22.924631       1 azure_routes.go:65] create: creating route. clusterName="jbariaskubetest" instance="k8s-master-21425678-0" cidr="10.244.1.0/24"
E0619 06:52:22.925220       1 routecontroller.go:160] Could not create route d099d06f-54b9-11e7-a827-000d3aa29c74 10.244.0.0/24 for node k8s-agentpool1-21425678-0 after 61.877009ms: network.SubnetsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Network/virtualNetworks/KubernetesCustomVNET' under resource group 'k8sresourcegroup' was not found."
I0619 06:52:22.925292       1 routecontroller.go:154] Creating route for node k8s-agentpool1-21425678-0 10.244.0.0/24 with hint d099d06f-54b9-11e7-a827-000d3aa29c74, throttled 600ns
I0619 06:52:22.925314       1 azure_routes.go:65] create: creating route. clusterName="jbariaskubetest" instance="k8s-agentpool1-21425678-0" cidr="10.244.0.0/24"
E0619 06:52:22.968368       1 routecontroller.go:160] Could not create route d1afc54f-54b9-11e7-a827-000d3aa29c74 10.244.1.0/24 for node k8s-master-21425678-0 after 43.793575ms: network.SubnetsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Network/virtualNetworks/KubernetesCustomVNET' under resource group 'k8sresourcegroup' was not found."
E0619 06:52:22.985946       1 routecontroller.go:160] Could not create route d099d06f-54b9-11e7-a827-000d3aa29c74 10.244.0.0/24 for node k8s-agentpool1-21425678-0 after 60.651288ms: network.SubnetsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Network/virtualNetworks/KubernetesCustomVNET' under resource group 'k8sresourcegroup' was not found."

One thing I noticed as well is the azure.json under /etc/kubernetes looks like this:
{
    "cloud":"AzurePublicCloud",
    "tenantId": “****************************”,
    "subscriptionId": “*****************************”,
    "aadClientId": “**********************”,
    "aadClientSecret": “*********************”,
    "resourceGroup": “K8sresourcegroup”,
    "location": "southeastasia",
    "subnetName": "KubernetesSubnet",
    "securityGroupName": "k8s-master-21425678-nsg",
    "vnetName": "KubernetesCustomVNET",
    "routeTableName": "k8s-master-21425678-routetable",
    "primaryAvailabilitySetName": "agentpool1-availabilitySet-21425678"
}

Not sure which part of the flow injects and builds this Azure.json file.
